daffodil-io/src/main/scala/org/apache/daffodil/io/InputSourceDataInputStream.scala

```diff
@@ -614,40 +614,44 @@ final class InputSourceDataInputStream private(val inputSource: InputSource)
     markPool.finalCheck
   }
 
+  /**
+   * Returns exactly One nChars-long string, or Nope.
+   *
+   * @param nChars
+   * @param finfo
+   * @return
+   */
   final def getString(nChars: Long, finfo: FormatInfo): Maybe[String] = {
     val startingBitPos = bitPos0b
-    val aligned = align(finfo.encodingMandatoryAlignmentInBits, finfo)
-    if (!aligned) {
-      Maybe.Nope
-    } else {
-      withLocalCharBuffer { lcb =>
-        val cb = lcb.getBuf(nChars)
-        val numDecoded = finfo.decoder.decode(this, finfo, cb)
-        if (numDecoded == nChars) {
-          Maybe(cb.flip.toString)
-        } else {
-          setBitPos0b(startingBitPos)
-          Maybe.Nope
-        }
+    withLocalCharBuffer { lcb =>
+      val cb = lcb.getBuf(nChars)
+      val numDecoded = finfo.decoder.decode(this, finfo, cb)
+      if (numDecoded == nChars) {
+        Maybe(cb.flip.toString)
+      } else {
+        setBitPos0b(startingBitPos)
+        Maybe.Nope
       }
```

Comment:
Code coverage says those lines are never called.  I looked for callers of `getString(nChars, finfo)` and the only place I found seems to be one line in [TextParser.scala](https://github.com/apache/daffodil/blob/55f638de74edfa400ce6cba1de19162069c9298a/daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/dfa/TextParser.scala#L57) which completely ignores the return value and doesn't check if Maybe.Nope is returned.  The new tests added below only call `getSomeString(nChars, finfo)`, not `getString(nChars, finfo)`.

Reply:
I'm going to consolidate, and just have getSomeString().